capstone-frontend/src/components/SearchBookModal.jsx

```diff
@@ -58,21 +58,11 @@ export class SearchBookModal extends Component {
 
   addBookToList = (book) => {
 
-    if (this.props.type === 'club') {
-      this.setState(
-        {
-          addedBooksIDs: [book.id],
-          addedBooks: [book]
-        }
-      )
-    }
-    else {
-      this.state.addedBooksIDs.push(book.id);
-      this.state.addedBooks.push(book);
+    this.state.addedBooksIDs.push(book.id);
+    this.state.addedBooks.push(book);
```

Comment:
In this case, instead of having this with the rerender below I think you can do something like this:
`this.setState({ addedBookIDs: [...this.state.addedBookIDs, book.id] })`

Taken from https://stackoverflow.com/questions/37435334/correct-way-to-push-into-state-array

Reply:
I can change this, but this function is solely used for the BookList implementation and didn't want to play with anything not related to my feature in case it alters how Mauro uses it. I can discuss with him

Reply:
Spoke with Mauro. We made the change and it works fine! Updating and merging.